src/main/java/com/csfrequency/firebase/devicestore/FirebaseDeviceStore.java

```diff
@@ -13,26 +13,31 @@
 import android.support.v4.content.LocalBroadcastManager;
 import android.util.Log;
 
+import com.google.android.gms.common.internal.Asserts;
+import com.google.android.gms.common.internal.Preconditions;
+import com.google.android.gms.tasks.Continuation;
 import com.google.android.gms.tasks.OnCompleteListener;
 import com.google.android.gms.tasks.Task;
+import com.google.android.gms.tasks.Tasks;
 import com.google.firebase.FirebaseApp;
 import com.google.firebase.auth.FirebaseAuth;
 import com.google.firebase.auth.FirebaseUser;
 import com.google.firebase.firestore.DocumentReference;
 import com.google.firebase.firestore.DocumentSnapshot;
+import com.google.firebase.firestore.FieldPath;
+import com.google.firebase.firestore.FieldValue;
 import com.google.firebase.firestore.FirebaseFirestore;
-import com.google.firebase.firestore.FirebaseFirestoreException;
-import com.google.firebase.firestore.Transaction;
+import com.google.firebase.firestore.SetOptions;
+import com.google.firebase.firestore.util.Assert;
```

Comment:
Uhhhhhh. I am surprised you can use this. This is not part of the the public API. Android Studio is supposed to not let you do this. I will try to figure out what is going on, but in the meantime, can you use something different (maybe Preconditions?).

Reply:
Ok, it looks like AS will let you import this but not use.

Reply:
Yes, good catch - I'd tried to use `Assert` and realised it wouldn't let me, but hadn't removed the import.  That's now sorted.